### DISCUSSION TITLE
Question about setting terminal colors for Midnight Commander (`mc`).

### DISCUSSION THREAD

**Discussion by juangiordana** [kovidgoyal/kitty#6221] [Q&A]
Question about setting terminal colors for Midnight Commander (`mc`).

I'm using Kitty with the Nightfox theme and I'm having trouble with the Midnight Commander (`mc`) colors being too bright and making the application hard to use.

I managed to set an alias that resets the terminal colors to their default values before `mc` starts and then reverts those changes back once the application execution completes.

```sh
# Midnight commander (cd to latest working directory).
# [[ ! -f /usr/lib/mc/mc.sh ]] || source /usr/lib/mc/mc.sh
if [[ -f /usr/lib/mc/mc.sh ]]; then
    source /usr/lib/mc/mc.sh
    aliascmd="kitty @ set-colors ${HOME}/.config/kitty/Default-colors.conf ;"
    aliascmd+="${aliases[mc]} ;"
    aliascmd+="kitty @ set-colors --reset"
    [[ "${TERM}" = "xterm-kitty" ]] && alias mc="${aliascmd}"
fi
```

I was looking for an option that resets the colors to their original defaults before launching the program but I couldn't find anything in the documentation.

The issue is that if I configure kitty to use a theme, the colors of that theme become the default ones and I can change any of those manually but I find that troublesome.

I also tried to load the colors from an empty file but that triggers an error:

```
$ touch empty.conf
$ kitty @ set-colors empty.conf
Traceback (most recent call last):
  File "/usr/bin/../lib/kitty/kitty/boss.py", line 685, in _execute_remote_command
    response = handle_cmd(self, window, pcmd, peer_id, self_window)
  File "/usr/bin/../lib/kitty/kitty/remote_control.py", line 220, in handle_cmd
    ans = c.response_from_kitty(boss, self_window or window, PayloadGetter(c, payload))
  File "/usr/bin/../lib/kitty/kitty/rc/set_colors.py", line 127, in response_from_kitty
    patch_color_profiles(colors, profiles, payload_get('configured'))
TypeError: argument 1 must be dict, not None
```

Also tried to to load the colors from the original file, that contains only comments, but that didn't produce any results (no errors though):

```
$ kitty @ set-colors /usr/share/doc/kitty/kitty.conf
```

I ended up manually extracting the default color scheme from the original `kitty.conf` file, saving it in my ~/.config/kitty/Default-colors.conf and then I'm manually doing the loading/unloading.

It would be nice to have an option like `kitty @ set-colors NONE` unless I'm missing something.

I look forward for your comments.
Thanks!

Edit: Corrected a mistake in the script in case anyone wants to use it.

**Comment by kovidgoyal** [comment]
Unless you are doing this over ssh, simply run

kitten themes default

On Thu, Apr 27, 2023 at 09:01:20PM -0700, Juan Giordana wrote:
> Question about setting terminal colors for Midnight Commander (`mc`).
> 
> I'm using Kitty with the Nightfox theme and I'm having trouble with the Midnight Commander (`mc`) colors being too bright and making the application hard to use.
> 
> I managed to set an alias that resets the terminal colors to their default values before `mc` starts and then reverts those changes back once the application execution completes.
> 
> ```sh
> # Midnight commander (cd to latest working directory).
> # [[ ! -f /usr/lib/mc/mc.sh ]] || source /usr/lib/mc/mc.sh
> if [[ -f /usr/lib/mc/mc.sh ]]; then
>     source /usr/lib/mc/mc.sh
>     if [[ "${TERM}" != "xterm-kitty" ]]; then
>         return
>     fi
> 
>     aliascmd="kitty @ set-colors ${HOME}/.config/kitty/Default-colors.conf ;"
>     aliascmd+="${aliases[mc]} ;"
>     aliascmd+="kitty @ set-colors --reset"
>     alias mc="${aliascmd}"
> fi
> ```
> 
> I was looking for an option that resets the colors to their original defaults before launching the program but I couldn't find anything in the documentation.
> 
> The issue is that if I configure kitty to use a theme, the colors of that theme become the default ones and I can change any of those manually but I find that troublesome.
> 
> I also tried to load the colors from an empty file but that triggers an error:
> 
> ```
> $ touch empty.conf
> $ kitty @ set-colors empty.conf
> Traceback (most recent call last):
>   File "/usr/bin/../lib/kitty/kitty/boss.py", line 685, in _execute_remote_command
>     response = handle_cmd(self, window, pcmd, peer_id, self_window)
>   File "/usr/bin/../lib/kitty/kitty/remote_control.py", line 220, in handle_cmd
>     ans = c.response_from_kitty(boss, self_window or window, PayloadGetter(c, payload))
>   File "/usr/bin/../lib/kitty/kitty/rc/set_colors.py", line 127, in response_from_kitty
>     patch_color_profiles(colors, profiles, payload_get('configured'))
> TypeError: argument 1 must be dict, not None
> ```
> 
> Also tried to to load the colors from the original file, that contains only comments, but that didn't produce any results (no errors though):
> 
> ```
> $ kitty @ set-colors /usr/share/doc/kitty/kitty.conf
> ```
> 
> I ended up manually extracting the default color scheme from the original `kitty.conf` file, saving it in my ~/.config/kitty/Default-colors.conf and then I'm manually doing the loading/unloading.
> 
> It would be nice to have an option like `kitty @ set-colors NONE` unless I'm missing something.
> 
> I look forward for your comments.
> Thanks!
> 
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/kovidgoyal/kitty/discussions/6221
> You are receiving this because you are subscribed to this thread.
> 
> Message ID: ***@***.***>

-- 
_____________________________________

Dr. Kovid Goyal 
https://www.kovidgoyal.net
https://calibre-ebook.com
_____________________________________

**Reply by juangiordana** [reply]
`kitten themes default` loads my currently configured theme (Nightfox) instead of the original.

That was one of the first options I've tried but with a capital D, because that's the name used inside the kitten.

I'm using kitty 0.28.1 on Arch.

**Reply by juangiordana** [reply]
`kitten themes -h` doesn't show that option either.

Also, it's all for my local setup.

**Reply by kovidgoyal** [reply]
Works for me. Steps I tried

1) kitten themes nightfox
2) run kitty, it uses nightfox theme, as expected
3) Inside the running kitty, execute kitten themes default
4) it switches to default theme, as expected
5) run kitten themes nightfox
6) goes back to nightfox, as expected.

**Reply by juangiordana** [reply]
Ah, that works, but my setup is different since I am sourcing Nightfox as the default in `~/.config/kitty/kitty.conf` and when I want to go to the original default it takes that as the default one.
